python/samples/demos/document_generator/custom_selection_strategy.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from typing import TYPE_CHECKING, ClassVar
+
+from opentelemetry import trace
+
+from semantic_kernel.agents.strategies.selection.selection_strategy import SelectionStrategy
+from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.azure_chat_prompt_execution_settings import (
+    AzureChatPromptExecutionSettings,
+)
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_chat_completion import OpenAIChatCompletion
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+if TYPE_CHECKING:
+    from semantic_kernel.agents import Agent
+    from semantic_kernel.contents.chat_message_content import ChatMessageContent
+
+
+@experimental_class
+class CustomSelectionStrategy(SelectionStrategy):
+    """A selection strategy that selects the next agent intelligently."""
+
+    NUM_OF_RETRIES: ClassVar[int] = 3
+
+    chat_completion_service: OpenAIChatCompletion
```

Comment:
Nit: Maybe use ChatCompletionClientBase if someone adapts this and doesn't notice this once

Reply:
I would try and test this sample with other models -- not all models will play nicely with our agent framework -- either Anthropic or Mistral doesn't like the ordering of messages once broadcast from the channel and breaks. I forget which one.